### DISCUSSION TITLE
How to query a vertex by a property of another vertex

### DISCUSSION THREAD

**Discussion by bCamba** [Gremlinq/ExRam.Gremlinq#780] [Q&A] [answered]
This is probably a very stupid question.
I have a ShipVertex with a property AccountId which is also the id of a vertex of type AccountVertex.

I tried something like this

```
_g.V<ShipVertex>()
            .Values(__ => __.AccountId)
            .As((__, accountId) => __
                .V<AccountVertex>(accountId))
```

but it does not work.

Also I know this query works in raw gremlin, but I am not sure this is the best way to do it and also I am not sure how to do it in Gremlinq.

`g.V().hasLabel("ShipVertex").values("AccountId").as("a").V().where(containing("a"))`

**Comment by danielcweber** [comment, marked as answer]
Try something like this instead:

```
_g.V<ShipVertex>()
    .Values(__ => __.AccountId)
    .As((__, accountId) => __
        .V<AccountVertex>()
        .Where(x => x.Id == accountId.Value))
```

There's no better way to express this IMO. Note, however, that this might not get optimized on your graph db system, and you could end up will a full db scan every time. You should measure response times.
